src/mantine-core/src/components/Stepper/Stepper.tsx

```diff
@@ -122,7 +127,7 @@ export const Stepper: StepperComponent = forwardRef<HTMLDivElement, StepperProps
       return acc;
     }, [] as React.ReactNode[]);
 
-    const content = filteredChildren[active]?.props?.children;
+    const content = filteredChildren[active]?.props?.children ?? completedStep?.props?.children;
```

Comment:
We should use `active` to check whether to show completed step or current step. If I'm not wrong, with current implementation it will show the completed step if one of the steps does not have content. 

The logic should be simple: `filteredChildren.length - 1 > active ? completed : current step`

Reply:
Ah yes indeed, I did not think of this use case :+1:

Reply:
Fixed, added the condition